pkg/expr/metrics/metrics.go

```diff
@@ -0,0 +1,123 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// ExprMetrics is a struct that contains all the metrics for an implementation of the expressions service
+// shared between multiple versions of expressions service, which are delineated by the subsystem string
+type ExprMetrics struct {
+	DSRequests                 *prometheus.CounterVec
+	ExpressionsQuerySummary    *prometheus.SummaryVec
+	SqlCommandDuration         *prometheus.HistogramVec
+	SqlCommandErrorCount       *prometheus.CounterVec
+	SqlCommandCellCountSummary *prometheus.SummaryVec
+}
+
+func newExprMetrics(subsystem string) *ExprMetrics {
+	return &ExprMetrics{
+		DSRequests: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "grafana",
+			Subsystem: subsystem,
+			Name:      "ds_queries_total",
+			Help:      "Number of datasource queries made via server side expression requests",
+		}, []string{"error", "dataplane", "datasource_type"}),
+
+		ExpressionsQuerySummary: prometheus.NewSummaryVec(
+			prometheus.SummaryOpts{
+				Namespace:  "grafana",
+				Subsystem:  subsystem,
+				Name:       "expressions_queries_duration_milliseconds",
+				Help:       "Expressions query summary",
+				Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+			},
+			[]string{"status"},
+		),
+
+		SqlCommandDuration: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: "grafana",
+			Subsystem: subsystem,
+			Name:      "sql_command_duration_seconds",
+			Help:      "Duration of SQL command execution",
+			Buckets:   prometheus.DefBuckets,
+		}, []string{"status"}),
+
+		SqlCommandErrorCount: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Namespace: "grafana",
+			Subsystem: subsystem,
+			Name:      "sql_command_errors_total",
+			Help:      "Total number of SQL command execution errors",
+		}, []string{}),
+
+		SqlCommandCellCountSummary: prometheus.NewSummaryVec(
+			prometheus.SummaryOpts{
+				Namespace:  "grafana",
+				Subsystem:  subsystem,
+				Name:       "sql_command_cell_count",
+				Help:       "Summary of the number of cells in SQL command execution",
+				Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
```

Comment:
Don't understand https://grafana.com/blog/2022/03/01/how-summary-metrics-work-in-prometheus/ https://prometheus.io/docs/practices/histograms/ yet, and if those objectives are maybe buckets? If so not sure those are good ones?